iwyu_tool_test.py

```diff
@@ -292,6 +292,21 @@ def test_fixup_compilation_db(self):
         entry = canonical[0]
         self.assertEqual(os.path.join(self.cwd, 'Test.cpp'), entry['file'])
 
+    def test_fixup_from_entry_dir(self):
+        """ Compilation database abs path is based on an entry's directory. """
+        compilation_db = [
+            {
+                "directory": os.path.join(self.cwd, "tests"),
```

Comment:
I think you can use a synthetic path here -- looks like `realpath` does nothing if the path doesn't exist.

Reply:
Much easier correlate `/home/user/foobar` here and `self.assertEqual('/home/user/foobar/Test.cpp', entry['file'])` in the assertion if they're spelled out.

Reply:
Done, indeed the test is fine with a static path.